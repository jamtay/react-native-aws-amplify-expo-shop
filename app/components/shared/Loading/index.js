import React from 'react';
import {Spinner} from 'native-base';
import {COLOURS} from '../../../styles/colours';

/**
 * Loading spinner to display when loading
 * @param isLoading
 */
const Loading = ({isLoading}) => {
  return isLoading ? (
    <Spinner testID="spinner" color={COLOURS.DARK_PINK} />
  ) : null;
};

export default Loading;
