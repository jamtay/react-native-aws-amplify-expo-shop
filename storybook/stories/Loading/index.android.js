import React from 'react';
import PropTypes from 'prop-types';
import Loading from '../../../app/components/shared/Loading';

export default function LoadingStory({isLoading}) {
  return <Loading isLoading={isLoading} />;
}

LoadingStory.defaultProps = {};

LoadingStory.propTypes = {
  isLoading: PropTypes.bool,
};
