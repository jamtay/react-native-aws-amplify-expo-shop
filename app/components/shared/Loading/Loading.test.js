import React from 'react';

import renderer from 'react-test-renderer';
import {render} from 'react-native-testing-library';
import Loading from './index';

describe('Loading component tests', () => {
  describe('snapshot test', () => {
    it('should match stored snapshot', () => {
      const tree = renderer.create(<Loading isLoading={true} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('render with properties', () => {
    it('should render with correct loading spinner if isLoading is true', async () => {
      const {getByTestId} = render(<Loading isLoading={true} />);
      expect(await getByTestId('spinner')).toBeTruthy();
    });

    it('should render with correct loading spinner if isLoading is false', async () => {
      const {queryByTestId} = render(<Loading isLoading={false} />);
      expect(await queryByTestId('spinner')).toBeFalsy();
    });

    it('should render with correct loading spinner if isLoading is not supplied', async () => {
      const {queryByTestId} = render(<Loading />);
      expect(await queryByTestId('spinner')).toBeFalsy();
    });
  });
});
