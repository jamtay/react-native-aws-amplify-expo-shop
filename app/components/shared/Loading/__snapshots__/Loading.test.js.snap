// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component tests snapshot test should match stored snapshot 1`] = `
<ActivityIndicator
  animating={true}
  color="violet"
  hidesWhenStopped={true}
  size="large"
  style={
    Object {
      "height": 80,
    }
  }
  testID="spinner"
/>
`;
